src/formatDate.js

```diff
@@ -8,7 +8,55 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[fromFormat.length - 1];
+  const joinElement = toFormat[toFormat.length - 1];
+
+  const arrOfDate = date.split(splitElement);
+  const newDateArr = [];
+
+  const varDay = 'DD';
+  const varMonth = 'MM';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === varDay) {
+      const index = toFormat.indexOf(varDay);
+      const day = arrOfDate[i];
+
+      newDateArr[index] = day;
+    } else if (fromFormat[i] === varMonth) {
+      const index = toFormat.indexOf(varMonth);
+      const month = arrOfDate[i];
+
+      newDateArr[index] = month;
```

Comment:
Similar to the previous comment, ensure that 'index' is not -1 before assigning to 'newDateArr[index]' to avoid potential errors.